### PR TITLE
Add support for Disconnected deployment on vSphere

### PR DESCRIPTION
**Pull request by dahorak**
It requires changes from [MR!795](https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/-/merge_requests/795) and also depends on fixes for OOD issue [ONEOCPDEPL-116](https://issues.redhat.com/browse/ONEOCPDEPL-116).

Signed-off-by: Daniel Horak <dahorak@redhat.com>

### CODE REVIEW

File: conf/deployment/vsphere/upi_1az_rhcos_vsan_3m_3w_disconnected.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+# This config will work on _vSphere DC-ECO, DC-CP, DC-PS and vSphere7 DC-CP_
```
Comment (vavuthu) [inline conf/deployment/vsphere/upi_1az_rhcos_vsan_3m_3w_disconnected.yaml:2]:
```suggestion
# This config will work on vSphere DC-ECO, DC-CP, DC-PS and vSphere7 DC-CP
```

**Comment by dahorak** [comment]
Last issue related to Flexy `hardware_version` version was fixed (in the related [MR!795](https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/-/merge_requests/795)).

File: ocs_ci/deployment/flexy.py [hunk 2]
```diff
@@ -160,6 +159,45 @@ def deploy_prereq(self):
...
+                cluster_name = config.ENV_DATA["cluster_name"]
+                cluster_domain = f".{cluster_name}.{base_domain}"
+                config.FLEXY["LAUNCHER_VARS"].update(
+                    {
+                        "http_proxy": config.ENV_DATA["disconnected_http_proxy"],
+                        "https_proxy": config.ENV_DATA.get(
+                            "proxy_https_proxy",
+                            config.ENV_DATA["disconnected_http_proxy"],
+                        ),
+                        "no_proxy": f"{cluster_domain},ocs-qe-proxy-dc-eco.usersys.redhat.com",
```
Comment (jimdevops19) [inline ocs_ci/deployment/flexy.py:177]: might this should be f"{cluster_domain}.ocs-qe-proxy-dc-eco.usersys.redhat.com" ?

Comment (dahorak) [reply ocs_ci/deployment/flexy.py:177]: It is a comma separated list of URLs - first URL is the cluster domain, the second is mirror registry URL, but it shouldn't be hardcoded here - I'll send a fix.

**Comment by dahorak** [comment]
In the last commit, I moved deployment related proxy configuration from `ENV_DATA` to `DEPLOYMENT` section and document the variables. And also removed hardcoded no_proxy value for disconnected deployment (replaced by `disconnected_no_proxy` configuration parameter).

**Review by vasukulkarni — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

**Pull request merged**